modules/xfem/src/userobjects/MeshCut3DUserObject.C

```diff
@@ -0,0 +1,1113 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "MeshCut3DUserObject.h"
+
+#include "XFEMFuncs.h"
+#include "MooseError.h"
+#include "libmesh/string_to_enum.h"
+#include "MooseMesh.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/edge_edge2.h"
+#include "libmesh/serial_mesh.h"
+
+template <>
+InputParameters
+validParams<MeshCut3DUserObject>()
+{
+  InputParameters params = validParams<GeometricCutUserObject>();
+  params.addRequiredParam<MeshFileName>("mesh_file", "Mesh file for XFEM geometric cuts");
+  params.addParam<Real>(
+      "size_control", 0, "Criterion for refining elements while growing the crack");
+  params.addParam<unsigned int>("n_step_growth", 0, "Number of steps for crack growth");
```

Comment:
You are right.  This is the number of steps to grow the mesh.  However, assuming XFEM and this code are fully integrated, this number of steps is exactly the same as the number of steps to change the cuts.  The input mesh serves as the initial cut.  The cut (which is also the mesh) will be updated based on stress intensity factors.  Do you still want me to remove this, and hard code a number in the C file?

Reply:
Since you're using this for testing, I'd say leave it in there.  We can remove it later on when we don't need it.  Since you have the capability to grow the mesh in your cutter already, we definitely want to test that.